#!/usr/bin/env python

PACKAGE = 'stair_detector_geo'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, str_t, double_t, bool_t, int_t

gen = ParameterGenerator()
gen.add("show_result",                 bool_t,   0, "where to show result",                                                                                  False)
gen.add("debug",                       bool_t,   0, "where to show debug",                                                                                   False)
gen.add("ignore_invalid",              bool_t,   0, "whether to ignore invalid pixel in depth images",                                                       False)
gen.add("fill_invalid",                bool_t,   0, "whether to fill the invalid pixels in depth images",                                                    False)
gen.add("use_laplacian",               bool_t,   0, "whether to use laplacian for edge detection",                                                           False)
gen.add("canny_low_threshold",         double_t, 0, "The canny low threshold",                                                                               20, 0, 100)
gen.add("canny_ratio",                 double_t, 0, "the canny ratio",                                                                                       3,  0, 10)
gen.add("canny_kernel_size",           int_t,    0, "the canny kernel size",                                                                                 1, 0, 3)
gen.add("hough_min_line_length",       double_t, 0, "Hough Transform: minimum line length",                                                                  30, 1, 200)
gen.add("hough_max_line_gap",          double_t, 0, "Hough Transform: maximum line gap",                                                                     20, 1, 200)
gen.add("hough_threshold",             double_t, 0, "Hough Transform: Accumulator threshold parameter. Only those lines are returned that get enough votes", 50, 0, 500)
gen.add("hough_rho",                   int_t,    0, "Hough Transform: Distance resolution of the accumulator in pixels.",                                    8, 0, 20)
gen.add("hough_theta",                 double_t, 0, "Hough Transform: Angle resolution of the accumulator in radians.",                                      1, 1, 10)
gen.add("filter_slope_hist_bin_width", double_t, 0, "Filter by slope histogram bin width",                                                                   20, 1, 60)
gen.add("filter_slope_bandwidth",      double_t, 0, "Filter by target angle and bandwidth",                                                                  10, 1, 60)
gen.add("merge_max_dist_diff",         double_t, 0, "Maximum distance differences when merge lines",                                                         20, 1, 200)
gen.add("merge_max_angle_diff",        double_t, 0, "Maximum angle differences when merge lines",                                                            30, 1, 50)
gen.add("merge_close_count",           int_t,    0, "How many pixel count needed to consider same line",                                                     10, 1, 30)
gen.add("minimum_line_num",            int_t,    0, "How many minimum line needed to consider valid stairs",                                                 5, 1, 20)
gen.add("max_valid_detection",         int_t,    0, "How many minimum potential image found to consider valid stairs",                                       10, 1, 20)
exit(gen.generate(PACKAGE, "stair_detector_geo_node", "StairDetectorGeo"))